envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyClustersFactory.kt

```diff
@@ -73,13 +73,20 @@ internal class EnvoyClustersFactory(
         return when (group) {
             is ServicesGroup -> group.proxySettings.outgoing.getServiceDependencies()
                 .mapNotNull {
-                    if (enableTlsForCluster(group, globalSnapshot, it.service)) {
-                        globalSnapshot.securedClusters.resources().get(it.service)
-                    } else {
-                        globalSnapshot.clusters.resources().get(it.service)
-                    }
+                    selectCluster(group, globalSnapshot, it.service)
+                }
+            is AllServicesGroup -> globalSnapshot.allServicesGroupsClusters
```

Comment:
The values from `globalSnapshot.allServicesGroupsClusters` are no longer used, so we can make it a Set of service-names

Reply:
Maybe we can just use `globalSnapshot.clusters.resources()` and use cluster name?